src/hooks/useGetPurposeMessageKeys.ts

```diff
@@ -0,0 +1,49 @@
+import { ConsentSelection } from '../types';
+
+import { useMemo } from 'preact/hooks';
+
+import { DefinedMessage } from '@transcend-io/internationalization';
+
+const CUSTOM_PURPOSE_MESSAGE_PREFIX = 'cm-ui.purpose';
+
+export interface PurposeMessages {
+  /** The lookup of purpose trackingTypes to message key */
+  purposeToMessageKey: Record<string, DefinedMessage>;
+  /** The order of all purposes */
+  orderOfPurposes: string[];
+}
+
+export const useGetPurposeMessageKeys = ({
+  consentSelection,
+  defaultPurposeToMessageKey,
+}: {
+  /** The configured airgap purpose types */
+  consentSelection: ConsentSelection;
+  /** The lookup of messages for default purpose types */
+  defaultPurposeToMessageKey: Record<string, DefinedMessage>;
+}): PurposeMessages => {
+  const purposeToMessageKey: Record<string, DefinedMessage> = useMemo(
+    () =>
+      // the purpose type is unique for the bundle
+      Object.keys(consentSelection ?? {}).reduce((allMessages, purposeType) => {
+        if (allMessages[purposeType]) {
+          return allMessages;
+        }
+        const customPurposeMessageLabel = `${CUSTOM_PURPOSE_MESSAGE_PREFIX}.${purposeType}`;
+        return {
+          ...allMessages,
+          [purposeType]: {
+            id: customPurposeMessageLabel,
+            defaultMessage: purposeType,
+            description: `Translatable name for custom purpose '${purposeType}'`,
+          } as DefinedMessage,
+        };
+      }, defaultPurposeToMessageKey as Record<string, DefinedMessage>),
+    [consentSelection, defaultPurposeToMessageKey],
+  );
+  const orderOfPurposes = useMemo(
+    () => Object.keys(purposeToMessageKey),
+    [purposeToMessageKey],
+  );
```

Comment:
Does this mean customers will be able to customize the order or purposes?

Reply:
the existing order of purposes is on purpose - would be great to maintain that if possible!

Reply:
no haha we just want to make sure the order is deterministic here

Reply:
@michaelfarrell76 ohhh yes i see - i'll update the logic to keep that in place

Reply:
would be nice to support this in the platform eventually, but i get thats harder

Reply:
![Screenshot 2023-08-10 at 4 16 53 PM](https://github.com/transcend-io/consent-manager-ui/assets/12694524/8fcbae7c-7a15-4a97-aa42-826096ae2290)

done